perception/euclidean_cluster/lib/voxel_grid_based_euclidean_cluster.cpp

```diff
@@ -124,6 +124,10 @@ bool VoxelGridBasedEuclideanCluster::cluster(
         &temporary_clusters.at(map[index]).data[cluster_data_size],
         &pointcloud_msg->data[i * point_step], point_step);
       cluster_data_size += point_step;
+      if (cluster_data_size == temporary_clusters.at(map[index]).data.size()) {
+        temporary_clusters.at(map[index])
+          .data.resize(temporary_clusters.at(map[index]).data.size() * 2);
```

Comment:
I think decide that the max size of output cluster (`cluster_data_size`) to be doubled of temporary_cluster is not reasonable. (runtime_error might be occurred here for some high dense pointcloud)
```suggestion
          .data.resize(max_cluster_size_ * point_step);
```


Or we can increment `cluster_data_size` and check it repeatedly.

Reply:
Here is a `for loop` that reallocates twice the space every time the space is full. When the space is full, it will twice the space again.I have tested using a ouster64, and the dense pointcloud is sometimes 50 times larger than the downsampled one. Even so, enough space can be allocated quickly.
If you use `max_cluster_size_`, it might as well just use the original code. If we add 10 points each time, this will result in multiple copies of the vector data.

Reply:
@beginningfan I understood. I have checked and it worked well. Thank you for your improvement.